lib/services/crawler/crawler.go

```diff
@@ -0,0 +1,153 @@
+package crawler
+
+import (
+	"database/sql"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"gorm.io/gorm"
+	"net/url"
+	"strings"
+
+	"github.com/markgravity/golang-ic/helpers/log"
+	"github.com/markgravity/golang-ic/lib/models"
+
+	"github.com/gocolly/colly/v2"
+	"github.com/gocolly/colly/v2/extensions"
+)
+
+type Crawler struct {
```

Comment:
@markgravity What do you think about separate this service into small services for small purposes only.

- GoogleSearchService: With a keyword, and returns HTML response
- GoogleParseService: With a HTML response, parse it to ParsingResult

Reply:
@suho I think this crawler is too small just 150 lines and the logic is not too complex, I suggest we do it when our logic is more complex in the future 🙏